actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchNotificationActor.java

```diff
@@ -0,0 +1,361 @@
+        (Map<String, String>) courseBatchObject.get(JsonKey.COURSE_ADDITIONAL_INFO);
+    Map<String, Object> requestMap = new HashMap<String, Object>();
+    requestMap.put(JsonKey.ORG_NAME, courseBatchObject.get(JsonKey.ORG_NAME));
+    requestMap.put(JsonKey.COURSE_LOGO_URL, additionalCourseInfo.get(JsonKey.COURSE_LOGO_URL));
+    requestMap.put(JsonKey.COURSE_NAME, additionalCourseInfo.get(JsonKey.COURSE_NAME));
+    requestMap.put(JsonKey.START_DATE, courseBatchObject.get(JsonKey.START_DATE));
+    requestMap.put(JsonKey.END_DATE, courseBatchObject.get(JsonKey.END_DATE));
+    requestMap.put(JsonKey.COURSE_ID, courseBatchObject.get(JsonKey.COURSE_ID));
+    requestMap.put(JsonKey.NAME, courseBatch.getName());
+    return requestMap;
+  }
+
+  /*
+   * @param Map<String, Object> data for populating user object
+   *
+   * @param operationType String for operation type
+   *
+   * @return map<String, String>
+   */
+  private Map<String, String> getUserData(Map<String, Object> data, String operationType) {
+    Map<String, String> user = new HashMap<String, String>();
+    user.put(JsonKey.FIRST_NAME, data.get(JsonKey.FIRST_NAME).toString());
+    user.put(JsonKey.EMAIL, data.get(JsonKey.EMAIL).toString());
+    if (operationType.equalsIgnoreCase(JsonKey.ADD)) {
+      user.put(JsonKey.SUBJECT, JsonKey.COURSE_INVITATION);
+    } else if (operationType.equalsIgnoreCase(JsonKey.REMOVE)) {
+      user.put(JsonKey.SUBJECT, JsonKey.UNENROLL_FROM_COURSE_BATCH);
+    }
+    return user;
+  }
+
+  /*
+   * This method is for sending e-mails for one at a time, EmailServiceActor will
+   * be called for sending email
+   *
+   * @param Map<String, String> user will have user data
+   *
+   * @param Map<String, Object> RequestMap for email data.
+   */
+  private void sendMail(Map<String, String> user, Map<String, Object> requestMap) {
+    requestMap.put(JsonKey.FIRST_NAME, user.get(JsonKey.FIRST_NAME));
+    requestMap.put(JsonKey.EMAIL_TEMPLATE_TYPE, user.get(JsonKey.EMAIL_TEMPLATE_TYPE));
+    requestMap.put(JsonKey.RECIPIENT_EMAILS, new String[] {user.get(JsonKey.EMAIL)});
+    requestMap.put(JsonKey.SUBJECT, user.get(JsonKey.SUBJECT));
+    requestMap.put(JsonKey.REQUEST, BackgroundOperations.emailService.name());
+    ActorRef ref = system.actorOf(props);
+    try {
+      Response response = emailServiceClient.sendMail(ref, requestMap);
+      sender().tell(response, self());
+      Response res = new Response();
+      res.setResponseCode(ResponseCode.OK);
+      sender().tell(res, self());
+    } catch (Exception e) {
+      sender().tell(e, self());
+      ProjectLogger.log(
+          "CourseBatchNotificationActor: sendMail using emailService client failed **** " + e,
+          LoggerEnum.ERROR);
+    }
+  }
+
+  /*
+   * This method takes CourseBatch object and returns list of participants
+   *
+   * @param CourseBatch object
+   *
+   * @returns list of users
+   */
+  private List<String> getParticipants(CourseBatch courseBatch) {
```

Comment:
This method is not necessary as the list of added / removed participants is passed.